math/uint_test.go

```diff
@@ -126,6 +126,19 @@ func (s *uintTestSuite) TestArithUint() {
 		n2 := uint64(rand.Uint32())
 		u2 := sdkmath.NewUint(n2)
 
+		if n1 >= n2 {
```

Comment:
this is covering only the happy path. In order to run into your check an see the panic, you would need an example where n1 < n2.  See my comment on the issue.

personal preference: this is a complex test already. Sub does not really fit well into the test data set. It would have been easier to cover the cases in a new unit test.